go/stats/counters.go

```diff
@@ -19,36 +19,85 @@ package stats
 import (
 	"bytes"
 	"fmt"
+	"strconv"
 	"strings"
 	"sync"
 	"sync/atomic"
+
+	"vitess.io/vitess/go/sync2"
 )
 
+// Counter is expvar.Int+Get+hook
+type Counter struct {
+	i    sync2.AtomicInt64
+	help string
+}
+
+// NewCounter returns a new Counter
+func NewCounter(name string, help string) *Counter {
+	v := &Counter{help: help}
+	if name != "" {
+		publish(name, v)
+	}
+	return v
+}
+
+// Add adds the provided value to the Counter
+func (v *Counter) Add(delta int64) {
```

Comment:
FWIW the prom API includes an `Inc()` method which seems nicer than `Add(1)`, especially if we follow through with the idea to prevent Add of a negative number.

Reply:
Added logging for now!